service/controller/defaultnamespace/resource/automationsa/create.go

```diff
@@ -0,0 +1,346 @@
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	pkgkey "github.com/giantswarm/rbac-operator/pkg/key"
+	"github.com/giantswarm/rbac-operator/pkg/project"
+	"github.com/giantswarm/rbac-operator/pkg/rbac"
+	"github.com/giantswarm/rbac-operator/service/controller/defaultnamespace/key"
+)
+
+// EnsureCreated Ensures that the automation service account is created in the default namespace,
+// and it has all the necessary RoleBindings and ClusterRoleBindings
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	namespace, err := key.ToNamespace(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	if !pkgkey.IsDefaultNamespace(namespace.Name) {
+		return nil
+	}
+
+	err = r.createAutomationServiceAccount(ctx, namespace.Name)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.createReadAllClusterRoleBindingToAutomationSA(ctx, namespace.Name)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.createWriteAllRoleBindingToAutomationSA(ctx, namespace.Name)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.createWriteOrganizationsClusterRoleBindingToAutomationSA(ctx, namespace.Name)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.createWriteClustersClusterRoleBindingToAutomationSA(ctx, namespace.Name)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.createWriteNodePoolsClusterRoleBindingToAutomationSA(ctx, namespace.Name)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.createWriteClientCertsClusterRoleBindingToAutomationSA(ctx, namespace.Name)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = r.createWriteSilencesClusterRoleBindingToAutomationSA(ctx, namespace.Name)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+// Ensures the 'automation' service account in the default namespace.
+func (r *Resource) createAutomationServiceAccount(ctx context.Context, namespace string) error {
+
+	automationSA := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: pkgkey.AutomationServiceAccountName,
+			Labels: map[string]string{
+				label.ManagedBy: project.Name(),
+			},
+		},
+	}
+
+	_, err := r.K8sClient().CoreV1().ServiceAccounts(namespace).Get(ctx, automationSA.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		r.Logger().LogCtx(ctx, "level", "info", "message", fmt.Sprintf("creating serviceaccount %#q in namespace %s", automationSA.Name, namespace))
+
+		_, err := r.K8sClient().CoreV1().ServiceAccounts(namespace).Create(ctx, automationSA, metav1.CreateOptions{})
+		if apierrors.IsAlreadyExists(err) {
+			// do nothing
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.Logger().LogCtx(ctx, "level", "info", "message", fmt.Sprintf("serviceaccount %#q in namespace %s has been created", automationSA.Name, namespace))
+
+	} else if err != nil {
+		return microerror.Mask(err)
+	} else {
```

Comment:
For other resources we use `createOrUpdate` functions so maybe we could have the same for service accounts?

Reply:
I created a `createOrUpdate` function for Service Accounts and put it in its own package.